### PR TITLE
[language] Fix error handling for non-ASCII input

### PR DESCRIPTION
**Pull request by bob-wilson**
The compiler was previously crashing when trying to handle the error loc because the list of source files was not updated when the input contains non-ASCII characters. Also tweak the error message.

## Motivation

Fix compiler crash

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Added a new test for this.

### CODE REVIEW

File: language/move-lang/tests/move_check/parser/invalid_character_non_ascii.move [hunk 1]
```diff
@@ -0,0 +1,6 @@
+address 0x0:
+
+// Check that we provide good error messages for non-ASCII characters.
+module Temp {
+   ф
```
Comment (sblackshear) [inline language/move-lang/tests/move_check/parser/invalid_character_non_ascii.move:5]: A test that checks that it works when the character is in a comment (probably the most likely scenario?) might also be useful.

Comment (bob-wilson) [reply language/move-lang/tests/move_check/parser/invalid_character_non_ascii.move:5]: At least for now, we intentionally disallow non-ASCII characters in comments (see https://github.com/libra/libra/pull/812), so I assume that by "it works" you mean that the compiler does not crash and produces a good error message. I added a test for that.

Comment (sblackshear) [reply language/move-lang/tests/move_check/parser/invalid_character_non_ascii.move:5]: Ah yes, that is indeed what I meant. Thanks for adding the test!

**Pull request closed without merging**